File: app/src/main/java/com/codemybrainsout/imageviewer/view/detail/DetailModule.java
package com.codemybrainsout.imageviewer.view.detail;

import android.content.Context;

import com.codemybrainsout.imageviewer.api.PhotoService;
import com.codemybrainsout.imageviewer.dagger.scope.ActivityScope;
import com.codemybrainsout.imageviewer.dagger.scope.ApplicationScope;
import com.codemybrainsout.imageviewer.model.Photo;

import dagger.Module;
import dagger.Provides;

/**
 * Created by ahulr on 09-06-2017.
 */

@Module
public class DetailModule {

    private Context context;

    public DetailModule(Context context) {
        this.context = context;
    }

    @ActivityScope
    @Provides
    Context context() {
        return context;
    }

}

File: app/src/main/java/com/codemybrainsout/imageviewer/view/detail/DetailModule.kt
package com.codemybrainsout.imageviewer.view.detail

import com.codemybrainsout.imageviewer.api.PhotoService
import dagger.Binds
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

/**
 * Created by ahulr on 09-06-2017.
 */
@Module
@InstallIn(ActivityComponent::class)
abstract class DetailModule {

    @Binds
    abstract fun bindActivity(activity: DetailActivity): DetailContract.View

    @Binds
    abstract fun bindPresenter(impl: DetailPresenter): DetailContract.Presenter
}